### PR TITLE
fix: enable connection to custom database name other than 'postgres'

### PR DESCRIPTION
**Pull request by w3b6x9**
## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

I can only connect to the `postgres` database when connecting via Supavisor pooler.

## What is the new behavior?

I can connect to another database other than `postgres` database.

## Additional context

This addresses the GH issue [Supavisor forces request to "postgres" database even when other database is specified #234](https://github.com/supabase/supavisor/issues/234).

While this solution enables connecting to other database, that's not `postgres` database, it presents other problems.

Here are some things that should be tackled alongside this change for completeness:

1. As Supavisor currently stands, this change enables listening to other database when the tenant has no existing local pool. However, if tenant wants to listen to other database then `postgres` database then it will default to other database. Whichever database a tenant connects to first will be the default one.
~2. Supavisor will continuously attempt to connect to a database even when that database doesn't exist. However, the client doesn't get the proper error feedback that database doesn't exist. See screen recording:~

Spun this out into its own issue: https://github.com/supabase/supavisor/issues/257

https://github.com/supabase/supavisor/assets/5532241/6b07f139-66be-4d7b-965a-bb769df9c037

### CODE REVIEW

File: docs/development/setup.md [hunk 1]
```diff
@@ -26,13 +26,14 @@ curl -X PUT \
...
-    "default_pool_size: 15,
+    "default_pool_size": 15,
     "users": [
       {
         "db_user": "postgres",
         "db_password": "postgres",
         "mode_type": "transaction",
-        "pool_checkout_timeout": 100
+        "pool_checkout_timeout": 100,
+        "pool_size": 10
```
Comment (w3b6x9) [inline docs/development/setup.md:36]: I was getting an error without this.

File: lib/supavisor_web/ws_proxy.ex [hunk 2]
```diff
@@ -61,6 +61,6 @@ defmodule SupavisorWeb.WsProxy do
   @spec connect_local() :: {:ok, port()} | {:error, term()}
   defp connect_local() do
     proxy_port = Application.fetch_env!(:supavisor, :proxy_port_transaction)
-    :gen_tcp.connect('localhost', proxy_port, [:binary, packet: :raw, active: true])
+    :gen_tcp.connect(~c"localhost", proxy_port, [:binary, packet: :raw, active: true])
```
Comment (abc3) [inline lib/supavisor_web/ws_proxy.ex:64]: I think we don't need sigils here, single quotes are a pretty native method for creating charlists

File: test/integration/proxy_test.exs [hunk 3]
```diff
@@ -170,10 +170,10 @@ defmodule Supavisor.Integration.ProxyTest do
     assert :httpc.request(
              "http://localhost:#{Application.get_env(:supavisor, :proxy_port_transaction)}"
            ) ==
-             {:ok, {{'HTTP/1.1', 204, 'OK'}, [], []}}
+             {:ok, {{~c"HTTP/1.1", 204, ~c"OK"}, [], []}}
```
Comment (abc3) [inline test/integration/proxy_test.exs:173]: the same about ~c

**Comment by abc3** [comment]
legend 🔥

File: lib/supavisor_web/ws_proxy.ex (hunk 2 above)
Comment (w3b6x9) [reply lib/supavisor_web/ws_proxy.ex:64]: this was result of `mix format`

File: test/integration/proxy_test.exs (hunk 3 above)
Comment (w3b6x9) [reply test/integration/proxy_test.exs:173]: this was result of `mix format`

File: lib/supavisor_web/ws_proxy.ex (hunk 2 above)
Comment (w3b6x9) [reply lib/supavisor_web/ws_proxy.ex]: reverted back

File: test/integration/proxy_test.exs (hunk 3 above)
Comment (w3b6x9) [reply test/integration/proxy_test.exs]: reverted back

**Review by abc3 — APPROVED** [review]

**Pull request merged**